src/charm.py

```diff
@@ -165,13 +177,13 @@ def _render_conf_file(self) -> str:
         rendered = template.render(
             mappers_path=str(self._mappers_dir_path),
             identity_schema_file_path=self._identity_schema_file_path,
-            default_browser_return_url=self.config.get("login_ui_url"),
+            default_browser_return_url=self._get_login_ui_endpoint_info("default_url"),
             public_base_url=self._domain_url,
-            login_ui_url=join(self.config.get("login_ui_url"), "login"),
-            error_ui_url=join(self.config.get("login_ui_url"), "oidc_error"),
+            login_ui_url=self._get_login_ui_endpoint_info("login_url"),
+            error_ui_url=self._get_login_ui_endpoint_info("error_url"),
```

Comment:
Kratos won't accept an empty value for a defined config option. So if any of the endpoints wasn't found and thus was returned as `None`, kratos would error out due to invalid config.
You could try overcoming it by modifying the template and not adding the config if the endpoint is `None`, like we do [here](https://github.com/canonical/kratos-operator/blob/main/templates/kratos.yaml.j2#L54) with the `oauth2_provider_url`.
However you need to test if kratos will start its service without any of the login-ui's endpoints.

Reply:
I updated the template